components/org.wso2.carbon.siddhi.editor.core/src/main/resources/web/editor/js/design-view/cron-generator/cron-generator.js

```diff
@@ -0,0 +1,771 @@
+                    errorMsg = "Numeric values between " + val + " and " + endVal;
+                } else {
+                    errorMsg = "Invalid Cron Expression"
+                }
+                return parseInt(month) >= val && parseInt(month) <= endVal;
+            });
+        };
+
+        var isValidDayOfMonthValue = function(dayOfMonth, dayOfWeek) {
+            if((dayOfMonth === '*' && dayOfWeek !== '*') || (dayOfMonth === '?' && dayOfWeek !== '?')) {
+                return true;
+            } else if(typeof dayOfMonth === 'string' && dayOfWeek === '?' && (dayOfMonth === 'LW' || dayOfMonth === 'L')){
+                return true;
+            } else if(typeof dayOfMonth === 'string' && dayOfWeek === '?' && dayOfMonth.includes('L')) {
+                if(dayOfMonth.includes(',') && dayOfWeek === '?'){
+                    return false;
+                }
+                if(dayOfMonth.includes('-') && dayOfWeek === '?') {
+                    var dayOfMonthRangeArr = dayOfMonth.split('-');
+                    var isLastDayIncludes = dayOfMonthRangeArr[0] === 'L' &&
+                                                        isValidateNumericValue([dayOfMonthRangeArr[1]], 1, 30);
+                    return isValidateNumericValue(dayOfMonthRangeArr, 1, 31) || isLastDayIncludes;
+                }
+            }
+            if (dayOfMonth.search(/[^\d-,*\/]/) !== -1) {
+                return false;
+            }
+            var list = dayOfMonth.split(",");
+            return list.every(function (dayOfMonth){
+                if(dayOfMonth.includes('/') && dayOfWeek === '?') {
+                    var startingDayOfMonthOptionArr = dayOfMonth.split('/');
+                    var isValidElements = (isValidateNumericValue([startingDayOfMonthOptionArr[0]], 1, 31) &&
+                                                        isValidateNumericValue([startingDayOfMonthOptionArr[1]], 1, 31));
+                    var isValidFirstElem = (startingDayOfMonthOptionArr[0] === '*' &&
+                                                        isValidateNumericValue([startingDayOfMonthOptionArr[1]], 1, 31));
+                    return isValidElements || isValidFirstElem;
+                } else if(dayOfMonth.includes('-') && dayOfWeek === '?') {
+                    var dayOfMonthRangeArr = dayOfMonth.split('-');
+
+                    return isValidateNumericValue(dayOfMonthRangeArr, 1, 31) &&
+                                    (parseInt(dayOfMonthRangeArr[0]) < parseInt(dayOfMonthRangeArr[1]));
+
+                } else if(typeof dayOfMonth === 'string' && dayOfWeek === '?') {
+                       return isValidateNumericValue([dayOfMonth], 1, 31);
+                }
+            });
+        };
+
+        var MONTH_LIST = ['jan', 'feb', 'mar', 'apr', 'may', 'jun', 'jul', 'aug', 'sep', 'oct', 'nov', 'dec'];
+
+        var isValidateStringValue = function(monthArr, dataArr) {
+            errorMsg = "Invalid Cron Expression";
```

Comment:
i do not see a place you use this error message or this function as intended to

Reply:
fixed in 0fb87f3